yt/yt/client/table_client/composite_compare.cpp

```diff
@@ -276,4 +308,112 @@ TFingerprint CompositeFarmHash(TYsonStringBuf value)
 
 ////////////////////////////////////////////////////////////////////////////////
 
+std::optional<TYsonString> TruncateCompositeValue(TYsonStringBuf value, i64 size)
+{
+    YT_ASSERT(value.GetType() == EYsonType::Node);
+
+    if (size <= 0) {
+        return {};
+    }
+
+    TMemoryInput valueIn(value.AsStringBuf());
+    TYsonPullParser valueParser(&valueIn, EYsonType::Node);
+
+    TString truncatedYson;
+    TStringOutput output(truncatedYson);
+    output.Reserve(std::min(size, std::ssize(value.AsStringBuf())));
```

Comment:
Don't you want to add a fast path `if (value.AsStringBuf().size() < size) return value` here?

Reply:
Idk. If the value contains a map inside it the returned values cannot be compared. However, such values cannot be compared with the original yson even if we truncated the map.

Reply:
Since we won't be saving an allocation here anyway, I think it is not necessary.